BrowserServicesKit/Sources/Suggestions/SuggestionProcessing.swift

```diff
@@ -26,14 +26,22 @@ public enum Platform {
 }
 
 /// Class encapsulates the whole ordering and filtering algorithm
-/// It takes query, history, bookmarks, and apiResult as input parameters
+/// It takes query, history, bookmarks, open tabs, and apiResult as input parameters
 /// The output is instance of SuggestionResult
-final class SuggestionProcessing {
+struct SuggestionProcessing {
+
+    // MARK: - Constants
+
+    static let maximumNumberOfSuggestions = 12
+    static let maximumNumberOfTopHits = 2
+    static let minimumNumberInSuggestionGroup = 5
 
     private let platform: Platform
+    private var isUrlIgnored: (URL) -> Bool
 
-    init(platform: Platform) {
+    init(platform: Platform, isUrlIgnored: @escaping (URL) -> Bool) {
         self.platform = platform
+        self.isUrlIgnored = isUrlIgnored
     }
 
     func result(for query: String,
```

Comment:
this function stays mostly the same: adding documentation for clarity and making it structurally match the windows implementation.